### ISSUE TITLE
Possible Starfield AVM Enum mismatches

### ISSUE THREAD

**Issue by deveris256** [Mutagen-Modding/Mutagen#573]
Hello!
I am working on a Starfield mod using Mutagen, writing it in C#.

When I sent my mod to people I know for them to check for possible mistakes, they noticed the AVM texture types (color, normal, rough, etc.) and AVM types (SimpleGroup, ComplexGroup) mismatch, e.g. `Normal` actually becoming `Rough` when mod is viewed in xEdit, and so on. I used the following enums to set the types where relevant:
```c#
ActorValueModulation.TextureTypeEnum
ActorValueModulation.TypeEnum
```

At the moment, I don't have access to xEdit, therefore I won't be able to test it properly. But, would be a good way to test it using:
- Creating AVM groups of all available types and then checking it in xEdit or somewhere else, and aligning Enum values if neccessary.

**Comment by Noggog** [comment]
I updated the TypeEnum to match:

xedit:
https://github.com/TES5Edit/TES5Edit/blob/dev-4.1.6/Core/wbDefinitionsSF1.pas#L9153

mutagen:
https://github.com/Mutagen-Modding/Mutagen/commit/95817d0bbe057e9fd6a37ed3e70eeb9a482cf4af


But the Texture type enum i didnt see a desync in the same way:

xedit:
https://github.com/TES5Edit/TES5Edit/blob/dev-4.1.6/Core/wbDefinitionsSF1.pas#L18836

mutagen:
https://github.com/Mutagen-Modding/Mutagen/blob/dev/Mutagen.Bethesda.Starfield/Records/Major%20Records/ActorValueModulation.cs#L13

Can you help narrow down the differences they were seeing in the TextureTypeEnum side of things?
It might be that xedit itself is still out of sync with the CK and that's what they're seeing?

**Comment by deveris256** [comment]
The mismatch is present clearly if one will open a Mutagen-created mod with AVM Texture Type set using `ActorValueModulation.TextureTypeEnum` in xEdit or other software. For example, if one will set the AVM Texture Type to `ActorValueModulation.TextureTypeEnum.Opacity` using Mutagen, then view the AVM Texture Type - it would not be 'Opacity' in xEdit, but rather other type. At least, in my experiences.

Aside, it is present in other Mutagen-related projects: For example, I have Starfield.esm processed into .json files using Spriggit. And, there is a mismatch too - Texture Types seem mismatched, the name of the record could contain `Rough` (e.g. `SimpleGroup_AVM_Razorback_FreestarRangersSidearm_Rough - 1CB04F_Starfield.esm.json/yaml`), but the AVM Texture Type is something irrelevant, like 'Opacity' in the case of 'Rough'.

**Comment by Noggog** [comment]
I will need more gritty details before I can adjust Texture Types. 

I looked at the raw hex for that record in question. It has a value of 2 for `MODT`:
![image](https://github.com/user-attachments/assets/8c13c4b4-6e04-4547-a873-f0ffdc764a34)

According to xEdit's own sourcecode.. that should be `Opacity`
![image](https://github.com/user-attachments/assets/f4814020-9af7-4c12-8783-d2b80d5c10cf)

Which is what Mutagen shows for that value:
![image](https://github.com/user-attachments/assets/8e660db4-6afb-4224-b57a-09acb9b5d88f)

 I did notice that xEdit's GUI shows differently:
![image](https://github.com/user-attachments/assets/542a0497-7d92-486f-97f7-af81a29d4d46)

But that doesn't line up with its own codebase (as shown above)...which makes me wonder if perhaps the latest GUI is stale and using older code?  Not sure what other reason there would be for xEdit's UI to differ from its own code.

Either way, I'll need more information on which is actually correct:  xEdit's UI?  Or xEdit's code?
Or perhaps there is something else at play?

**Comment by Noggog** [comment]
I'm not at a computer that can open it up, but if someone wants to crack open the record in the CK and show what it has (and also, what the dropdown has in general)... that would be useful in diagnosing the desync as well

**Comment by user19990313** [comment]
Well, the wbFlags in xEdit is for bit-packed flags, therefore the numbers are not the integer values, they are the exponent...
https://github.com/TES5Edit/TES5Edit/blob/dev-4.1.6/Core/wbInterface.pas#L9859
So Rough should be 2^1 instead of 1, Opacity should be 2^2 instead of 2. Additionally, 2^0=1 is apparently named "Model" flag in CK, so both Normal and Opacity flag in mutagen would be considered as Rough in xEdit and CK because 2^1 bit is set for both.
Usually the flag value comments are written as {0x01} {0x02} {0x04} etc, but someone got lazy for this one.

**Comment by Noggog** [comment]
@user19990313 good catch!

Should be fixed in commit 993b267b4bfef99e4e195a10c70d70c9454cad6a

And pushed out as a nightly release.   Let me know if you have issues using the nightly @deveris256 and ill push out a legit version instead

Also feel free to reopen if I missed anything else
